WordPress/gutenberg
# Components refactor `Mobile` to ignore `exhuastive-deps`

## What?
Refactor the Mobile components to ignore the `exhaustive-deps` eslint rule.

## Why?
Because the Components squad generally lacks the React Native expertise and tools/environments to effectively refactor these components, we'll defer to the native team to handle when they're ready.

## How?
Added inline ignore comments to all `exhaustive-deps` warnings in the `Mobile` component directory

## Testing Instructions

1. From your local Gutenberg directory, run `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/mobile`
2. Confirm that the linter returns no errors
